internal/pkg/template/diff/diff.go

```diff
@@ -88,7 +89,80 @@ func isYAMLLeaf(node *yaml.Node) bool {
 }
 
 func parseSequence(from, to *yaml.Node) (map[string]*Node, error) {
-	return nil, nil
+	toSeq, fromSeq := make([]yaml.Node, len(to.Content)), make([]yaml.Node, len(from.Content))
+	if err := to.Decode(&toSeq); err != nil {
+		return nil, err
+	}
+	if err := from.Decode(&fromSeq); err != nil {
+		return nil, err
+	}
+	seen, err := compare(fromSeq, toSeq)
+	if err != nil {
+		return nil, err
+	}
+	var index int
+	children := make(map[string]*Node)
+	for _, occurrence := range seen {
+		if occurrence.inTo == occurrence.inFrom {
```

Comment:
If I'm following correctly, we're assuming that that Sequences behave as `sets` is that correct? (i.e. the order in which a node appears is not relevant)

I wonder if this is always a safe assumption 💭 Can we think of any counter examples? If not perhaps we can drop a comment explaining that here :)

Reply:
You are right 😭  Some intrinsic functions like `!Sub` takes an ordered list. These are all I've found so far after I search through our codebase. I can't think of any counter examples so far for "real" CFN fields (will update this comment if any one comes up). Good catch!